### PR TITLE
Preemption fails when host or vnode is requested at chunk level

### PR DESCRIPTION
**Pull request by arungrover**
#### Describe Bug or Feature
PBS scheduler while trying to preempt a job tries to limit the number of preemption candidates by checking the reason why a high priority job could not run and then cull the preemptible jobs to only those jobs who can help in running the high priority job.
While doing so, it takes special care about non-consumable resource vnode and host. The issue while checking these resources was that it was assumed these two are job wide resources and thus they are requested for the whole job. But this assumption is incorrect, one can request a multinode job to have one chunk running on a specific vnode or host.
Scheduler was crashing because of this wrong assumption.

#### Describe Your Change
In this change, we try to retain the value of vnode/host because of which the job could not run in the error structure. We store it in arg2 of error structure. Then in cull_preemptible_jobs, instead of searching though all the chunks and finding out which chunk requested the resource that caused the job to not run, we now use the arg2 member of error structure that stores the name of the vnode/host because of which the job could not run.

#### Link to Design Doc
NA

#### Attach Test and Valgrind Logs/Output
[test_preemption_before.txt](https://github.com/openpbs/openpbs/files/5633185/test_preemption_before.txt)
[test_preemption.txt](https://github.com/openpbs/openpbs/files/5633188/test_preemption.txt)

### CODE REVIEW

File: src/scheduler/check.cpp [hunk 1]
```diff
@@ -1192,13 +1192,19 @@ check_avail_resources(schd_resource *reslist, resource_req *reqlist,
 				if (!compare_non_consumable(res, resreq)) {
 					fail = 1;
 					if (err != NULL) {
+						char *requested;
```
Comment (vstumpf) [inline src/scheduler/check.cpp:1195]: Can you make this a `const char *`? Eventually I'd like `res_to_str_r` return a `const char *`,

**Review by bhroam — APPROVED** [review]
LGTM

**Review by vstumpf — CHANGES_REQUESTED** [review]
LGTM

**Review by vstumpf — COMMENTED** [review]
LGTM

**Review by vstumpf — APPROVED** [review]
LGTM (third time's the charm)

**Pull request merged**